test/cctest/test_environment.cc

```diff
@@ -66,7 +68,32 @@ TEST_F(EnvironmentTest, EnvironmentWithESMLoader) {
       "})()");
 }
 
+class RedirectStdErr {
+ public:
+  explicit RedirectStdErr(const char* filename) : filename_(filename) {
+    fflush(stderr);
+    fgetpos(stderr, &pos_);
+    fd_ = dup(fileno(stderr));
+    freopen(filename_, "w", stderr);
+  }
+
+  ~RedirectStdErr() {
+    fflush(stderr);
+    dup2(fd_, fileno(stderr));
+    close(fd_);
+    remove(filename_);
+    clearerr(stderr);
+    fsetpos(stderr, &pos_);
+  }
+
+ private:
+  int fd_;
+  fpos_t pos_;
+  const char* filename_;
+};
+
 TEST_F(EnvironmentTest, EnvironmentWithNoESMLoader) {
+  RedirectStdErr redirect_scope("environment_test.log");
```

Comment:
Can you add a comment explaining why this is necessary?

Reply:
No problem, I'll add a comment :+1: